src/components/TodoList.tsx

```diff
@@ -0,0 +1,102 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React from 'react';
+import { Todo } from '../types/Todo';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+
+interface TodoListProps {
+  filteredTodos: Todo[];
+  handleUpdateTodoStatus: (id: number) => void;
+  handleRemoveTodo: (id: number) => void;
+  handleSelectedTodoChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  handleSubmitForm: (e: React.FormEvent<HTMLFormElement>, todo: Todo) => void;
+  editingTodoId: number[];
+  selectedTodo: Todo | null;
+  setSelectedTodo: (todo: Todo | null) => void;
+}
+export function TodoList({
+  filteredTodos,
+  handleUpdateTodoStatus,
+  handleRemoveTodo,
+  handleSelectedTodoChange,
+  handleSubmitForm,
+  editingTodoId,
+  selectedTodo,
+  setSelectedTodo,
+}: TodoListProps) {
+  return (
+    <TransitionGroup component={null}>
+      {filteredTodos.map(todo => (
+        <CSSTransition
+          key={todo.id}
+          timeout={300}
+          classNames="item"
+          unmountOnExit
+          appear
+          enter
+          exit
+        >
+          <div
+            data-cy="Todo"
+            className={`todo ${todo.completed ? 'completed' : ''}`}
+          >
+            <label className="todo__status-label">
+              <input
+                data-cy="TodoStatus"
+                type="checkbox"
+                className="todo__status"
+                checked={todo.completed}
+                onClick={() => handleUpdateTodoStatus(todo.id)}
+              />
+            </label>
+
+            {!selectedTodo ? (
+              <span
+                onDoubleClick={() => setSelectedTodo(todo)}
```

Comment:
Also, instead of using this `<span>` on this map, try isolating the code of one todo to a component `Todo` in a way that here you'll have:

![image](https://github.com/user-attachments/assets/78c26d41-5ccf-4087-b7ea-f6ea0150bc68)


Not exactly like the picture, it's just an idea for you to work on, create more components. The goal is to have components that do one thing, and just one thing. This component `TodoList` should handle the list, and only the list, not each todo individually

Reply:
I've tried to change to your suggestion, but i've noticed that it is written in a way that make more sense, i "avoided" handle using "Ids" in this case, because the user only can  handle one edit per time.  Thus, adding the Todo in a selectedTodo when the user make a double click in this span, trigger the toggle to the form. Then, in the form, is dealing with the other cases that the user click outside of the form, etc..

Reply:
@italomagno I'm sorry, I sent you a direct message on our platform. That's all on me, I made a mistake while reviewing your code, and used the wrong parameters to review it. I'll correct it the next time.